app/forms/field_handlers/number_handler.py

```diff
@@ -70,7 +66,7 @@ def get_field(self) -> Union[DecimalField, IntegerField]:
 
     def get_field_references(
         self,
-    ) -> dict[str, Union[bool, Optional[AnswerValueTypes]]]:
+    ) -> dict[str, Union[bool, Union[ValueSourceEscapedTypes, ValueSourceTypes]]]:
```

Comment:
The nested `Union` is redundant.
```suggestion
    ) -> dict[str, Union[bool, ValueSourceEscapedTypes, ValueSourceTypes]]:
```